tests/test_coco_format.py

```diff
@@ -836,6 +847,320 @@ def test_can_pickle(self):
                 compare_datasets_strict(self, source, parsed)
 
 
+class CocoExtractorTests(TestCase):
+    @mark_requirement(Requirements.DATUM_ERROR_REPORTING)
+    def test_can_report_unexpected_file(self):
+        with TestDir() as test_dir:
+            with self.assertRaisesRegex(DatasetImportError, "JSON file"):
+                CocoInstancesExtractor(test_dir)
+
+    @mark_requirement(Requirements.DATUM_ERROR_REPORTING)
+    def test_can_report_missing_item_field(self):
+        for field in ["id", "file_name"]:
+            with self.subTest(field=field):
+                with TestDir() as test_dir:
+                    ann_path = osp.join(test_dir, "ann.json")
+                    anns = {
+                        "images": [
+                            {
+                                "id": 5,
+                                "width": 10,
+                                "height": 5,
+                                "file_name": "a.jpg",
+                            }
+                        ],
+                        "annotations": [],
+                        "categories": [],
+                    }
```

Comment:
Most of these tests are very similar; I think it would be more readable to factor out the common parts into a single function that takes one parameter (a function that mangles the dataset) and returns the resulting exception. Then the individual tests could just look like this:

```python
def mangle(anns):
    del anns["images"][0][field]

exc = self._test_mangled_annotations(mangle)
self.assertIsInstance(capture.exception.__cause__, MissingFieldError)
self.assertEqual(capture.exception.__cause__.name, field)
```

Reply:
Agree about the common part extraction, I'll check if it's possible.

Reply:
Done

Reply:
You extracted a shared annotation template (which is certainly a big improvement), but there's still a lot of duplication in the structure of the tests. How about factoring that out too?

Reply:
How do you see that?

Reply:
Maybe, but it seem to save 1 line of code per test.